src/Harmony/Signer.cpp

```diff
@@ -88,10 +344,142 @@ Data Signer::rlpNoHash(const Transaction &transaction, const bool include_vrs) c
     return RLP::encodeList(encoded);
 }
 
+template <typename Directive>
+Data Signer::rlpNoHash(const Staking<Directive> &transaction, const bool include_vrs) const
```

Comment:
If `Signer` becomes too long, you can try to move encoding move to separate class/files